packages/contracts/contracts/core/plugin/PluginUUPSUpgradeable.sol

```diff
@@ -2,39 +2,45 @@
 
 pragma solidity 0.8.10;
 
-import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 
 import {DaoAuthorizableUpgradeable} from "../component/DaoAuthorizableUpgradeable.sol";
 import {IDAO} from "../IDAO.sol";
 import {PluginUpgradeable} from "./PluginUpgradeable.sol";
 
 /// @title PluginUUPSUpgradeable
+/// @author Aragon Association - 2022
 /// @notice An abstract, upgradeable contract to inherit from when creating a plugin being deployed via the UUPS pattern (see [ERC-1822](https://eips.ethereum.org/EIPS/eip-1822)).
 abstract contract PluginUUPSUpgradeable is PluginUpgradeable, UUPSUpgradeable {
+    /// @notice The [ERC-165](https://eips.ethereum.org/EIPS/eip-165) interface ID of the contract.
     bytes4 public constant PLUGIN_UUPS_UPGRADEABLE_INTERFACE_ID =
         type(PluginUUPSUpgradeable).interfaceId;
+
+    /// @notice The ID of the permission required to call the `_authorizeUpgrade` function.
     bytes32 public constant UPGRADE_PERMISSION_ID = keccak256("UPGRADE_PERMISSION");
 
     // NOTE: When adding new state variables to the contract, the size of `_gap` has to be adapted below as well.
 
-    /// @dev Used to check the permissions within the upgradability pattern implementation of OZ
-    function _authorizeUpgrade(address) internal virtual override auth(UPGRADE_PERMISSION_ID) {}
-
-    /// @inheritdoc PluginUpgradeable
-    function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
+    /// @notice Checks if an interface is supported by this or its parent contract.
+    /// @param _interfaceId The ID of the interace.
+    /// @return bool Returns true if the interface is supported.
+    function supportsInterface(bytes4 _interfaceId) public view virtual override returns (bool) {
         return
-            interfaceId == PLUGIN_UUPS_UPGRADEABLE_INTERFACE_ID ||
-            super.supportsInterface(interfaceId);
+            _interfaceId == PLUGIN_UUPS_UPGRADEABLE_INTERFACE_ID ||
+            _interfaceId == PLUGIN_UPGRADEABLE_INTERFACE_ID ||
+            super.supportsInterface(_interfaceId);
     }
 
-    /// @notice used to check the current base logic contract proxy delegates to.
-    /// @return address the address of current base logic contract.
-    function getImplementationAddress() public view returns (address) {
-        return _getImplementation();
+    /// @notice Returns the address of the implementation contract the UUPS proxy is pointing to.
+    /// @return implementation The address of the implementation contract.
+    function getImplementationAddress() public view returns (address implementation) {
+        implementation = _getImplementation();
```

Comment:
it's small but directly returning instead of assigning it to `implementation` would save some small amount of gas

Reply:
Done

Reply:
my bad. optimization corrects this and it isn't more costly